core/src/main/java/com/linecorp/armeria/client/ClientFactoryBuilder.java

```diff
@@ -242,15 +247,30 @@ private void channelOptions(Map<ChannelOption<?>, Object> newChannelOptions) {
     }
 
     /**
-     * Disables the verification of server's key certificate chain. This method is a shortcut for:
-     * {@code tlsCustomizer(b -> b.trustManager(InsecureTrustManagerFactory.INSTANCE))}.
+     * Disables the verification of server's TLS certificate chain. If you want to disable verification for
+     * only specific hosts, use {@link #tlsNoVerifyHosts(String...)}.
      * <strong>Note:</strong> You should never use this in production but only for a testing purpose.
      *
      * @see InsecureTrustManagerFactory
      * @see #tlsCustomizer(Consumer)
      */
     public ClientFactoryBuilder tlsNoVerify() {
-        tlsCustomizer(b -> b.trustManager(InsecureTrustManagerFactory.INSTANCE));
+        checkState(insecureHosts.isEmpty(), "tlsNoVerify() and tlsNoVerifyHosts() are mutually exclusive.");
+        tlsNoVerifySet = true;
+        return this;
+    }
+
+    /**
+     * Disables the verification of server's TLS certificate chain for specific hosts. If you want to disable
+     * all verification, use {@link #tlsNoVerify()} .
+     * <strong>Note:</strong> You should never use this in production but only for a testing purpose.
+     *
+     * @see IgnoreHostsTrustManager
```

Comment:
Need to remove this line because `IgnoreHostsTrustManager` is not part of the public API.